### ISSUE TITLE
The audio waveform isn't displayed sometimes if an audio component has a change event attached

### ISSUE THREAD

**Issue by jorgegarcia-ea** [gradio-app/gradio#7490]
### Describe the bug

When loading different audio files into a gradio audio component (independently of sample rate or length), sometimes the waveform isn't displayed. Additionally, audio playback isn't possible.

This only seems to happen when there is a change event attached to the audio component. If I remove it, the issue isn't present.

It's a blocking issue for me because I need to use audio change events in my app. 

Many thanks in advance!

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr

def on_audio_input_changed(audio_input):
    return gr.Button(interactive=True) if audio_input else gr.Button(interactive=False)

if __name__ == "__main__":
    with gr.Blocks() as app:
        audio_input = gr.Audio(label="Audio input", type='filepath')
        button = gr.Button(value="Click me")

        audio_input.change(on_audio_input_changed, audio_input, button)

    app.launch(inbrowser=True)
```

### Screenshot

![image](https://github.com/gradio-app/gradio/assets/150125846/e0f379f6-16e1-41eb-96c3-ab09550d80b0)

### Logs

```shell
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Exception in callback _ProactorBasePipeTransport._call_connection_lost(None)
handle: <Handle _ProactorBasePipeTransport._call_connection_lost(None)>
Traceback (most recent call last):
  File "C:\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Python311\Lib\asyncio\proactor_events.py", line 165, in _call_connection_lost
    self._sock.shutdown(socket.SHUT_RDWR)
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
```

### System Info

```shell
System and setup details: Windows 10, latest Chrome update, gradio 4.19.1
```

### Severity

Blocking usage of gradio

**Comment by abidlabs** [comment]
Hi @jorgegarcia-ea I can't repro this issue. Can you share the audio files that the issue is happening with? I can only think that its related to the files themselves since all of my files work

**Comment by jorgegarcia-ea** [comment]
[Fireworks-Fireworks-Launch-.zip](https://github.com/gradio-app/gradio/files/14355029/Fireworks-Fireworks-Launch-.zip)

Thank you for the reply @abidlabs. I can repro the issue with this file. The bit depth is 24-bit in this one, don't know if that is relevant.

**Comment by abidlabs** [comment]
Hi @jorgegarcia-ea I can't reproduce this issue even with your audio file:

<img width="1312" alt="image" src="https://github.com/gradio-app/gradio/assets/1778297/7e5084cc-e2ac-4072-bb55-14d118b60af3">

Do you see this when issue when you try running in other environments, such as [Colab notebook](https://colab.google/)?

**Comment by jorgegarcia-ea** [comment]
Thank you again for looking into this. I was able to reproduce the issue on a different computer and also when running in a Colab notebook.

![image](https://github.com/gradio-app/gradio/assets/150125846/e5ecac06-0c67-4039-86c8-67a42f545a92)

For the other computer, here is the log I've got when the issue occurs (this is the same error I've been seeing).

```
(gradio_test) C:\Users\jorge\Desktop>python Temp\gradio_test.py
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
C:\Users\jorge\Desktop\gradio_test\lib\site-packages\gradio\processing_utils.py:392: UserWarning: Trying to convert audio automatically from int32 to 16-bit int format.
  warnings.warn(warning.format(data.dtype))
Exception in callback _ProactorBasePipeTransport._call_connection_lost(None)
handle: <Handle _ProactorBasePipeTransport._call_connection_lost(None)>
Traceback (most recent call last):
  File "C:\Python39\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Python39\lib\asyncio\proactor_events.py", line 162, in _call_connection_lost
    self._sock.shutdown(socket.SHUT_RDWR)
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
ERROR:    Exception in ASGI application
  + Exception Group Traceback (most recent call last):
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 192, in __call__
  |     await response(scope, wrapped_receive, send)
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 264, in __call__
  |     await wrap(partial(self.listen_for_disconnect, receive))
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\anyio\_backends\_asyncio.py", line 678, in __aexit__
  |     raise BaseExceptionGroup(
  | exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 412, in run_asgi
    |     result = await app(  # type: ignore[func-returns-value]
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    |     return await self.app(scope, receive, send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\fastapi\applications.py", line 1054, in __call__
    |     await super().__call__(scope, receive, send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\applications.py", line 123, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
    |     raise exc
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
    |     await self.app(scope, receive, _send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 193, in __call__
    |     response_sent.set()
    |   File "C:\Python39\lib\contextlib.py", line 137, in __exit__
    |     self.gen.throw(typ, value, traceback)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\_utils.py", line 93, in collapse_excgroups
    |     raise exc
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 260, in wrap
    |     await func()
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 217, in stream_response
    |     return await super().stream_response(send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 254, in stream_response
    |     await send({"type": "http.response.body", "body": b"", "more_body": False})
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 161, in _send
    |     await send(message)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 516, in send
    |     output = self.conn.send(event=h11.EndOfMessage())
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 512, in send
    |     data_list = self.send_with_data_passthrough(event)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 545, in send_with_data_passthrough
    |     writer(event, data_list.append)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 67, in __call__
    |     self.send_eom(event.headers, write)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 96, in send_eom
    |     raise LocalProtocolError("Too little data for declared Content-Length")
    | h11._util.LocalProtocolError: Too little data for declared Content-Length
    +------------------------------------

During handling of the above exception, another exception occurred:

  + Exception Group Traceback (most recent call last):
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\_utils.py", line 87, in collapse_excgroups
  |     yield
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 193, in __call__
  |     response_sent.set()
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\anyio\_backends\_asyncio.py", line 678, in __aexit__
  |     raise BaseExceptionGroup(
  | exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Exception Group Traceback (most recent call last):
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 192, in __call__
    |     await response(scope, wrapped_receive, send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 264, in __call__
    |     await wrap(partial(self.listen_for_disconnect, receive))
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\anyio\_backends\_asyncio.py", line 678, in __aexit__
    |     raise BaseExceptionGroup(
    | exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
    +-+---------------- 1 ----------------
      | Traceback (most recent call last):
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 412, in run_asgi
      |     result = await app(  # type: ignore[func-returns-value]
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
      |     return await self.app(scope, receive, send)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\fastapi\applications.py", line 1054, in __call__
      |     await super().__call__(scope, receive, send)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\applications.py", line 123, in __call__
      |     await self.middleware_stack(scope, receive, send)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
      |     raise exc
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
      |     await self.app(scope, receive, _send)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 193, in __call__
      |     response_sent.set()
      |   File "C:\Python39\lib\contextlib.py", line 137, in __exit__
      |     self.gen.throw(typ, value, traceback)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\_utils.py", line 93, in collapse_excgroups
      |     raise exc
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 260, in wrap
      |     await func()
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 217, in stream_response
      |     return await super().stream_response(send)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 254, in stream_response
      |     await send({"type": "http.response.body", "body": b"", "more_body": False})
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 161, in _send
      |     await send(message)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 516, in send
      |     output = self.conn.send(event=h11.EndOfMessage())
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 512, in send
      |     data_list = self.send_with_data_passthrough(event)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 545, in send_with_data_passthrough
      |     writer(event, data_list.append)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 67, in __call__
      |     self.send_eom(event.headers, write)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 96, in send_eom
      |     raise LocalProtocolError("Too little data for declared Content-Length")
      | h11._util.LocalProtocolError: Too little data for declared Content-Length
      +------------------------------------

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 412, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\fastapi\applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
    raise exc
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 193, in __call__
    response_sent.set()
  File "C:\Python39\lib\contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\_utils.py", line 93, in collapse_excgroups
    raise exc
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 260, in wrap
    await func()
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 217, in stream_response
    return await super().stream_response(send)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 254, in stream_response
    await send({"type": "http.response.body", "body": b"", "more_body": False})
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 161, in _send
    await send(message)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 516, in send
    output = self.conn.send(event=h11.EndOfMessage())
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 512, in send
    data_list = self.send_with_data_passthrough(event)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 545, in send_with_data_passthrough
    writer(event, data_list.append)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 67, in __call__
    self.send_eom(event.headers, write)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 96, in send_eom
    raise LocalProtocolError("Too little data for declared Content-Length")
h11._util.LocalProtocolError: Too little data for declared Content-Length
ERROR:    Exception in ASGI application
  + Exception Group Traceback (most recent call last):
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 192, in __call__
  |     await response(scope, wrapped_receive, send)
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 264, in __call__
  |     await wrap(partial(self.listen_for_disconnect, receive))
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\anyio\_backends\_asyncio.py", line 678, in __aexit__
  |     raise BaseExceptionGroup(
  | exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 412, in run_asgi
    |     result = await app(  # type: ignore[func-returns-value]
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    |     return await self.app(scope, receive, send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\fastapi\applications.py", line 1054, in __call__
    |     await super().__call__(scope, receive, send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\applications.py", line 123, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
    |     raise exc
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
    |     await self.app(scope, receive, _send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 193, in __call__
    |     response_sent.set()
    |   File "C:\Python39\lib\contextlib.py", line 137, in __exit__
    |     self.gen.throw(typ, value, traceback)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\_utils.py", line 93, in collapse_excgroups
    |     raise exc
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 260, in wrap
    |     await func()
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 217, in stream_response
    |     return await super().stream_response(send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 254, in stream_response
    |     await send({"type": "http.response.body", "body": b"", "more_body": False})
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 161, in _send
    |     await send(message)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 516, in send
    |     output = self.conn.send(event=h11.EndOfMessage())
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 512, in send
    |     data_list = self.send_with_data_passthrough(event)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 545, in send_with_data_passthrough
    |     writer(event, data_list.append)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 67, in __call__
    |     self.send_eom(event.headers, write)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 96, in send_eom
    |     raise LocalProtocolError("Too little data for declared Content-Length")
    | h11._util.LocalProtocolError: Too little data for declared Content-Length
    +------------------------------------

During handling of the above exception, another exception occurred:

  + Exception Group Traceback (most recent call last):
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\_utils.py", line 87, in collapse_excgroups
  |     yield
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 193, in __call__
  |     response_sent.set()
  |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\anyio\_backends\_asyncio.py", line 678, in __aexit__
  |     raise BaseExceptionGroup(
  | exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Exception Group Traceback (most recent call last):
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 192, in __call__
    |     await response(scope, wrapped_receive, send)
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 264, in __call__
    |     await wrap(partial(self.listen_for_disconnect, receive))
    |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\anyio\_backends\_asyncio.py", line 678, in __aexit__
    |     raise BaseExceptionGroup(
    | exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
    +-+---------------- 1 ----------------
      | Traceback (most recent call last):
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 412, in run_asgi
      |     result = await app(  # type: ignore[func-returns-value]
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
      |     return await self.app(scope, receive, send)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\fastapi\applications.py", line 1054, in __call__
      |     await super().__call__(scope, receive, send)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\applications.py", line 123, in __call__
      |     await self.middleware_stack(scope, receive, send)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
      |     raise exc
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
      |     await self.app(scope, receive, _send)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 193, in __call__
      |     response_sent.set()
      |   File "C:\Python39\lib\contextlib.py", line 137, in __exit__
      |     self.gen.throw(typ, value, traceback)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\_utils.py", line 93, in collapse_excgroups
      |     raise exc
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 260, in wrap
      |     await func()
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 217, in stream_response
      |     return await super().stream_response(send)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 254, in stream_response
      |     await send({"type": "http.response.body", "body": b"", "more_body": False})
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 161, in _send
      |     await send(message)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 516, in send
      |     output = self.conn.send(event=h11.EndOfMessage())
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 512, in send
      |     data_list = self.send_with_data_passthrough(event)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 545, in send_with_data_passthrough
      |     writer(event, data_list.append)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 67, in __call__
      |     self.send_eom(event.headers, write)
      |   File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 96, in send_eom
      |     raise LocalProtocolError("Too little data for declared Content-Length")
      | h11._util.LocalProtocolError: Too little data for declared Content-Length
      +------------------------------------

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 412, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\fastapi\applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 186, in __call__
    raise exc
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 193, in __call__
    response_sent.set()
  File "C:\Python39\lib\contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\_utils.py", line 93, in collapse_excgroups
    raise exc
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 260, in wrap
    await func()
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\base.py", line 217, in stream_response
    return await super().stream_response(send)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\responses.py", line 254, in stream_response
    await send({"type": "http.response.body", "body": b"", "more_body": False})
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\starlette\middleware\errors.py", line 161, in _send
    await send(message)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 516, in send
    output = self.conn.send(event=h11.EndOfMessage())
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 512, in send
    data_list = self.send_with_data_passthrough(event)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_connection.py", line 545, in send_with_data_passthrough
    writer(event, data_list.append)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 67, in __call__
    self.send_eom(event.headers, write)
  File "C:\Users\jorge\Desktop\gradio_test\lib\site-packages\h11\_writers.py", line 96, in send_eom
    raise LocalProtocolError("Too little data for declared Content-Length")
h11._util.LocalProtocolError: Too little data for declared Content-Length
```

**Comment by jorgegarcia-ea** [comment]
Some additional information in case it helps: I've downgraded gradio to 4.17.0 and the issue isn't present in my environment.

**Comment by abidlabs** [comment]
Can you share the Colab notebook @jorgegarcia-ea?

**Comment by jorgegarcia-ea** [comment]
Hi @abidlabs, please find it attached.

[Gradio_test.zip](https://github.com/gradio-app/gradio/files/14521115/Gradio_test.zip)

**Comment by abidlabs** [comment]
Hi @jorgegarcia-ea sorry can you link to the Colab notebook instead of zipping it?

**Comment by jorgegarcia-ea** [comment]
@abidlabs https://colab.research.google.com/drive/1fRxXfsxZ4KwTl-wEg2n_CF7rVJuO_X-a?usp=sharing this is on my org account. Please let me know if you can't request access. Hope it helps!

**Comment by abidlabs** [comment]
Just requested access

**Comment by abidlabs** [comment]
Hi @jorgegarcia-ea I'm sorry but I can't repro this even with your Colab notebook: 

<img width="1361" alt="image" src="https://github.com/gradio-app/gradio/assets/1778297/7c1473de-3f20-4024-8577-22d3fcffa9e7">

What browser / OS are you using?

**Comment by abidlabs** [comment]
Going to close as we can't repro this issue at all, sorry about that.

**Comment by Corginyan** [comment]
I ran into the same problem today. And I think I figured out what's going on. Take a look at this fragment:

https://github.com/gradio-app/gradio/blob/7d714699ae9c47daabbd48a55f61e990c5bba1d7/js/audio/interactive/InteractiveAudio.svelte#L214-L215

It forces the same file to be uploaded to the same location twice. This would be fine if the widget didn't also try to retrieve the file twice during the process. A race condition occurs between the get and post requests, and the larger the record, the less orderly the order becomes. Ultimately, I suspect that there's an occasional attempt to read a file that hasn't yet been fully written, which causes the error.
